arraycontext/container/dataclass.py

```diff
@@ -32,14 +32,22 @@
 """
 
 from collections.abc import Mapping, Sequence
-from dataclasses import Field, fields, is_dataclass
-from typing import Union, get_args, get_origin
+from dataclasses import fields, is_dataclass
+from typing import NamedTuple, Union, get_args, get_origin
 
 from arraycontext.container import is_array_container_type
 
 
 # {{{ dataclass containers
 
+class _Field(NamedTuple):
```

Comment:
Any reason to make this a `NamedTuple` and not a dataclass? I'm not even sure why/how to decide between the two.

Reply:
Hm, not really.. I tend to use them for smaller private things, where I don't need default values, special inits, inheritance, etc.

I would expect them to be a bit faster and lighter if you don't need the extra features, but haven't really checked :\ Some more informed points: https://stackoverflow.com/a/51673969